tools/duplicate_function_eliminator.py

```diff
@@ -0,0 +1,355 @@
+
+import os, sys, subprocess, multiprocessing, re, string, json, shutil, logging, traceback
+import shared
+from js_optimizer import *
+
+DUPLICATE_FUNCTION_ELIMINATOR = path_from_root('tools', 'eliminate-duplicate-functions.js')
+
+def minify_shell(js, js_engine, shell, equivalentfn_hash_info=None):
+  shell = shell.replace('0.0', '13371337') # avoid uglify doing 0.0 => 0
```

Comment:
this horrible hack (my fault for writing it in `js_optimizer.py` ;) ) should not be needed, since we run before doubles are finalized now